packages/server-wallet/src/protocols/process-ledger-queue.ts

```diff
@@ -26,9 +27,33 @@ import {
 
 export type ProtocolState = {
   fundingChannel: ChannelState;
-  channelsPendingRequest: ChannelState[];
+  channelsRequestingFunds: ChannelState[];
+  channelsReturningFunds: ChannelState[];
 };
 
+const retrieveFundsFromClosedChannels = (
+  original: SimpleAllocation,
+  channelsReturningFunds: ChannelState[]
+): SimpleAllocation => ({
+  ...original,
+  allocationItems: channelsReturningFunds.reduce(
+    (allocationItems, channel) =>
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      (channel.supported!.outcome as SimpleAllocation).allocationItems
+        .reduce((allocationItems, {destination, amount}) => {
+          const idx = allocationItems.findIndex(to => destination === to.destination);
+          return idx > -1
+            ? _.update(allocationItems, idx, to => ({
+                destination,
+                amount: BN.add(amount, to.amount),
+              }))
+            : [...allocationItems, {destination, amount}];
+        }, allocationItems)
+        .filter(item => item.destination !== channel.channelId),
+    original.allocationItems
+  ),
+});
```

Comment:
This is kinda complicated. Curious what folks would prefer I do with this to make it easier to grok? Basically it is a map-reduce de-allocating funds into a single ledger outcome.